### DISCUSSION TITLE
GitHub Copilot could not connect to server. Extension activation failed: "connect ETIMEDOUT"

### DISCUSSION THREAD

**Discussion by jackhu-bme** [community/community#13887] [Copilot Conversations]
This happens on my AI lab server and does not happen on my personal macbook. I guess this happens for network settings but my AI lab server can be accessed both from school local internet and the Internet.But I really don't know how to solve it. Plus, ping command on the server is not permitted: "ping: sendmsg: operation not permitted", and I guess this is related to the problem.

**Comment by jackhu-bme** [comment]
What's more, it's OK to use copilot with pycharm locally, but it's not the case when I use the school AI lab server.

**Reply by 1664234934** [reply]
一样我的也是提示这个，不知道你解决了没有？

**Reply by jackhu-bme** [reply]
vscode没有解决，现在改用pycharm做远程开发了。我猜是因为pycharm插件可以调用本地机器的网络，本地上面我的网连github相关网站的都没问题。pycharm将本地代码同步到远程；vscode将远程代码显示出来，远程机器的网络相关可能有问题。我看到twitter上面有相关讨论有人宣称解决了，但是我没有推特账号因此看不了解决的评论蛮烦的。你可以去试着搜搜。

**Comment by murulidharab6** [comment]
i have the same error it works for a while and stops working, seems some stability issue on GHC?

**Comment by the-gsk** [comment]
It seems like your AI lab server might have some network restrictions or configurations that are causing GitHub Copilot to fail to connect to the server. Since you're unable to use the ping command on the server, it's likely that there are restrictions in place that are preventing certain network operations.

Here are a few steps you can take to troubleshoot and potentially resolve the issue:

1. Check Firewall Settings: Ensure that the firewall settings on your AI lab server allow outgoing connections to the necessary GitHub servers. GitHub Copilot might be trying to connect to specific servers that are being blocked by the firewall.

2. Proxy Settings: If your AI lab server is behind a proxy, make sure that the proxy settings are configured correctly. GitHub Copilot might require certain proxy settings to connect to GitHub servers.

3. Network Configuration: Verify that the network configuration on your AI lab server is correctly set up. Sometimes misconfigured network settings can cause connectivity issues.

4. DNS Resolution: Check if DNS resolution is working correctly on your AI lab server. GitHub Copilot needs to resolve GitHub's domain names to IP addresses to establish connections. If DNS resolution is failing, it could cause connection issues.

5. Contact IT Support: If you're unable to resolve the issue on your own, consider reaching out to your school's IT support team or the administrators of the AI lab server. They might be able to provide insights or assistance in resolving the network connectivity issues.

By troubleshooting these aspects, you might be able to identify and resolve the problem with GitHub Copilot's connectivity on your AI lab server.
